frontend/src/component/common/Sticky/Sticky.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+    HTMLAttributes,
+    ReactNode,
+    useContext,
+    useEffect,
+    useRef,
+    useState,
+} from 'react';
+import { StickyContext } from './StickyContext';
+import { styled } from '@mui/material';
+
+const StyledSticky = styled('div', {
+    shouldForwardProp: (prop) => prop !== 'top',
+})<{ top?: number }>(({ theme, top }) => ({
+    position: 'sticky',
+    zIndex: theme.zIndex.sticky - 100,
+    ...(top !== undefined
+        ? {
+              '&': {
+                  top,
+              },
+          }
+        : {}),
+}));
+
+interface IStickyProps extends HTMLAttributes<HTMLDivElement> {
+    children: ReactNode;
+}
+
+export const Sticky = ({ children, ...props }: IStickyProps) => {
```

Comment:
can we add some tests for this component since so many other things will depend on it

Reply:
Tried addressing this here: https://github.com/Unleash/unleash/pull/5088/commits/1eae0131cda178c3fc41c304d64f933691d3e5c1